### PR TITLE
add xyz.kusma.aurora

### PR DESCRIPTION
**Pull request by kusma**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

> All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.

I'm not really sure about this one. The application is using [BASS](www.un4seen.com/bass.html), which is proprietary software. I can't really find wording in the license that restricts redistribution, but also not something that explicitly allows it. But it allows non-commercial use, and redistribution is kind of a requirement for use.

So my interpretation is that this should be OK, but I'm not feeling 100% confident. Please advice!

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: :-(

but when you just repack a binary...

File: xyz.kusma.aurora.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+app-id: xyz.kusma.aurora
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: aurora.sh
+modules:
+- name: glslang
+  buildsystem: cmake
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:8]:
```suggestion
  buildsystem: cmake-ninja
```

File: xyz.kusma.aurora.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  sources:
+  - type: git
+    url: https://github.com/KhronosGroup/glslang.git
+    tag: 7.10.2984
+- name: glfw
+  buildsystem: cmake
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:14]:
```suggestion
  buildsystem: cmake-ninja
```

File: xyz.kusma.aurora.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  config-opts:
+  - "-DBUILD_SHARED_LIBS=1"
+  - "-DGLFW_BUILD_TESTS=0"
+  - "-DGLFW_BUILD_DOCS=0"
+  sources:
+  - type: git
+    url: https://github.com/glfw/glfw.git
+    tag: 3.3.6
+- name: glm
+  buildsystem: cmake
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:24]:
```suggestion
  buildsystem: cmake-ninja
```

File: xyz.kusma.aurora.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    sha256: f41379682f9ada94ea7b34fe86bf9ee00935a3147be41b6569c9605a53e438fd
+  build-commands:
+  - CXXFLAGS=-std=c++03 make -j
+  - make
+  - install -m644 Source/FreeImage.h /app/include
+  - install -m755 libfreeimage-3.18.0.so /app/lib
+  - ln -sf libfreeimage-3.18.0.so /app/lib/libfreeimage.so.3
+  - ln -sf libfreeimage.so.3 /app/lib/libfreeimage.so
+- name: assimp
+  buildsystem: cmake
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:46]:
```suggestion
  buildsystem: cmake-ninja
```

File: xyz.kusma.aurora.yaml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  - install -m644 xyz.kusma.aurora.desktop /app/share/applications/
+- name: appdata-file
+  buildsystem: simple
+  sources:
+  - type: file
+    path: xyz.kusma.aurora.appdata.xml
+  build-commands:
+  - mkdir -p /app/share/metainfo/
+  - install -m644 xyz.kusma.aurora.appdata.xml /app/share/metainfo/
+finish-args:
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:104]: all this section is usually *above* the modules

File: xyz.kusma.aurora.yaml [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+- "--socket=x11"
+- "--device=dri"
+- "--socket=pulseaudio"
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:107]: you still need `--share=ipc` for X11

File: xyz.kusma.aurora.yaml [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  - mkdir -p /app/share/applications/
+  - install -m644 xyz.kusma.aurora.desktop /app/share/applications/
+- name: appdata-file
+  buildsystem: simple
+  sources:
+  - type: file
+    path: xyz.kusma.aurora.appdata.xml
+  build-commands:
+  - mkdir -p /app/share/metainfo/
+  - install -m644 xyz.kusma.aurora.appdata.xml /app/share/metainfo/
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:103]:
```suggestion
  - install -Dm644 -t /app/share/metainfo xyz.kusma.aurora.appdata.xml 
```

File: xyz.kusma.aurora.yaml [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+  - install -m755 x64/libbass.so /app/lib
+- name: demo
+  buildsystem: meson
+  config-opts:
+  - "--buildtype=release"
+  sources:
+  - type: git
+    url: https://github.com/kusma/even-laster-engine.git
+    branch: aurora-linux
+- name: aurora
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:73]: all of these modules can be collapsed in the main module. You don't need 4 different

File: xyz.kusma.aurora.yaml [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+  - install -m 644 bass.h /app/include
+  - install -m755 x64/libbass.so /app/lib
+- name: demo
+  buildsystem: meson
+  config-opts:
+  - "--buildtype=release"
+  sources:
+  - type: git
+    url: https://github.com/kusma/even-laster-engine.git
+    branch: aurora-linux
```
Comment (hfiguiere) [inline xyz.kusma.aurora.yaml:72]: you need an explicit commit

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: xyz.kusma.aurora.desktop [hunk 11]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=Aurora
+Comment="Aurora" by Excess
+Exec=/app/bin/aurora.sh
```
Comment (hfiguiere) [inline xyz.kusma.aurora.desktop:5]:
```suggestion
Exec=aurora.sh
```

File: flathub.json (hunk 1 above)
Comment (kusma) [reply flathub.json:2]: Yeah, it's sad. The bass24-linux.zip also comes with an x86 binary, but I haven't figured out how to wire that up in the build system yet. I'm sure it's possible, though.

BASS has no support for Linux on other architectures. In the longer run, I want to get rid of this dependency, and use something open source, but I have been unable to find something that fits my needs so far. And writing a replacement is a bit more work than I currently have the time for...

File: xyz.kusma.aurora.yaml (hunk 7 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml:107]: Hmm, OK? It seems to work fine without for me, but I'm on Wayland.

File: xyz.kusma.aurora.yaml (hunk 10 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml:72]: Ah, yeah. Good point.

File: xyz.kusma.aurora.yaml (hunk 7 above)
Comment (hfiguiere) [reply xyz.kusma.aurora.yaml:107]: it's not a failure mode, it is a slow mode.

File: xyz.kusma.aurora.yaml (hunk 2 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 4 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 3 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 5 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 9 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml:78]: Good idea, fixed.

File: xyz.kusma.aurora.yaml (hunk 7 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 6 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.yaml (hunk 8 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Good suggestion, thanks!

File: xyz.kusma.aurora.yaml (hunk 10 above)
Comment (kusma) [reply xyz.kusma.aurora.yaml]: Fixed

File: xyz.kusma.aurora.desktop (hunk 11 above)
Comment (kusma) [reply xyz.kusma.aurora.desktop:5]: Fixed, thanks.

**Comment by kusma** [comment]
Hmm, I get a warning when installing saying "Warning: Failed to parse appdata: element 'content_rating' requires attribute 'type'". However, having an empty `content_rating`-tag was [suggested by the appstream docs](https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-content_rating)...

**Comment by barthalion** [comment]
bot, build xyz.kusma.aurora

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/xyz.kusma.aurora.appdata.xml: FAILED:
• tag-missing           : <release> required
Validation of files failed
```

**Comment by kusma** [comment]
Ugh, I just removed that, because I don't really do versioned releases of my demos. I guess I need to add one just for the Flatpak, then.

**Comment by barthalion** [comment]
Make sure to use `org.freedesktop.appstream-glib` from Flathub. We carry some patches that change the default behaviour.

**Comment by kusma** [comment]
OK, passes there now. I added the version 0.1.0, because, well... I don't really have a reasonable version number.

**Pull request closed without merging**